src/v/kafka/server/handlers/fetch.cc

```diff
@@ -77,8 +78,12 @@ static ss::future<read_result> read_from_partition(
     if (unlikely(!lso)) {
         co_return read_result(lso.error());
     }
+    auto maybe_start_ofs = co_await part.sync_effective_start();
+    if (!maybe_start_ofs) {
+        co_return read_result(map_topic_error_code(maybe_start_ofs.error()));
+    }
```

Comment:
I think this should be part of `replicated_partition::validate_fetch_offset`, this way we would validate the start offset with updated value, here it is only returned like this to the client